docs/source/topics/backends/snmp.rst

```diff
@@ -0,0 +1,149 @@
+
+************
+SNMP backend
+************
+
+SNMP
+====
+
+SNMP (Simple Network Management Protocol) is a network protocol very useful for retrieving info from a device.
+All the information is retrieved by using codes called MIBs. All MIBs have a tree-like structure, every main information is the root and by adding more detail to the info
+the tree gains more depth.
+Obviously, by getting the smallest MIB which is "1" or simply " . " one can get all the tree.
+
+The base SNMP backend contains the following methods (some internal methods are not documented and are subject to change in the future):
+
++--------------+------------------------------------------------------------------------------------------------------------------------------------------+
+| **to_dict**  | Returns a dict containing monitoring information depending on the type of the device.                                                    |
+|              | It follows the `NetJSON Devicemonitoring <https://netjson.org/rfc.html#name-devicemonitoring>`_ spec                                     |
++--------------+------------------------------------------------------------------------------------------------------------------------------------------+
+| **to_json**  | Calls the `to_dict` method and returns a JSON string of the dict                                                                         |
++--------------+------------------------------------------------------------------------------------------------------------------------------------------+
+| **validate** | Checks if connection with the device is working and raises `NetengineError` in case something is wrong                                   |
++--------------+------------------------------------------------------------------------------------------------------------------------------------------+
+
+Initializing an SNMP backend class requires the following arguments:
+
++---------------+---------------------------------------------------------------------+
+| **host**      | Management ip or hostname of the device                             |
++---------------+---------------------------------------------------------------------+
+| **community** | Community string for the SNMP connection. Default value is 'public' |
++---------------+---------------------------------------------------------------------+
+| **agent**     | Agent string for the SNMP connection                                |
++---------------+---------------------------------------------------------------------+
+| **port**      | Port for the SNMP connection. Default value is `161`                |
++---------------+---------------------------------------------------------------------+
```

Comment:
Are these parameters all mandatory?
Is there any default value for any parameter?
Let's make this explicit.
There's no need to write this table, sphinx has automations we can use to autodocument these things.
Please use it!
Refer to the netjsonconfig documentation for examples:
http://netjsonconfig.openwisp.org/en/latest/backends/openwrt.html#initialization

I mean this:

![Screenshot from 2021-08-19 17-31-32](https://user-images.githubusercontent.com/841044/130152817-1eb02414-a2d1-4d49-9b33-fe8e30ab2c5c.png)

See the source for more info: https://raw.githubusercontent.com/openwisp/netjsonconfig/master/docs/source/backends/openwrt.rst

The doc text is embedded in a docstring in the method itself.

This is a good practice, the docstring is also shown in some rich shells (bpython should show it, not sure about ipython).